src/components/v2/appDetails/AppDetails.component.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useState, useEffect } from 'react';
+import K8ResourceComponent from './k8Resource/K8Resource.component';
+import './appDetails.scss';
+import { ReactComponent as K8ResourceIcon } from '../../../assets/icons/ic-object.svg';
+import { ReactComponent as LogAnalyzerIcon } from '../../../assets/icons/ic-logs.svg';
+import { ReactComponent as Cross } from '../../../assets/icons/ic-close.svg';
+import LogAnalyzerComponent from './logAnalyzer/LogAnalyzer.component';
+import { NavLink, Route, Switch } from 'react-router-dom';
+import { useRouteMatch, Redirect, useParams, useHistory } from 'react-router';
+import { URLS } from '../../../config';
+import AppDetailsStore, { AppDetailsTabs } from './appDetails.store';
+import { useSharedState } from '../utils/useSharedState';
+import { ApplicationObject, AppStreamData, NodeType } from './appDetails.type';
+import NodeDetailComponent from './k8Resource/nodeDetail/NodeDetail.component';
+import Tippy from '@tippyjs/react';
+import IndexStore from './index.store';
+import EnvironmentStatusComponent from './sourceInfo/environmentStatus/EnvironmentStatus.component';
+import EnvironmentSelectorComponent from './sourceInfo/EnvironmentSelector.component';
+import SyncErrorComponent from './SyncError.component';
+import { Progressing, useEventSource } from '../../common';
+
+const AppDetailsComponent = () => {
+    const params = useParams<{ appId: string; envId: string; nodeType: string }>();
+    const { path, url } = useRouteMatch();
+    const [applicationObjectTabs] = useSharedState(
+        AppDetailsStore.getAppDetailsTabs(),
+        AppDetailsStore.getAppDetailsTabsObservable(),
+    );
+    const [loading, setLoading] = useState(true);
+    const history = useHistory();
+    const [streamData, setStreamData] = useState<AppStreamData>(null);
+    const appDetails = IndexStore.getAppDetails();
```

Comment:
we are not using state and observable.... when environment changes this may become problem as syncSSE uses environmentName, which can change in case of application details where same application is deployed to multiple environments

Reply:
This is outside the scope of v1